lib/membrane_h264_plugin/parser/nalu_parser.ex

```diff
@@ -3,42 +3,65 @@ defmodule Membrane.H264.Parser.NALuParser do
   A module providing functionality of parsing a stream of binaries, out of which each
   is a payload of a single NAL unit.
   """
+
+  alias Membrane.H264.Parser
   alias Membrane.H264.Parser.{NALu, NALuTypes}
-  alias Membrane.H264.Parser.NALuParser.SchemeParser
-  alias Membrane.H264.Parser.NALuParser.Schemes
+  alias Membrane.H264.Parser.NALuParser.{SchemeParser, Schemes}
+
+  @annexb_prefix_code <<0, 0, 0, 1>>
 
   @typedoc """
   A structure holding the state of the NALu parser.
   """
   @opaque t :: %__MODULE__{
-            scheme_parser_state: SchemeParser.t()
+            scheme_parser_state: SchemeParser.t(),
+            input_stream_structure: Parser.stream_structure(),
+            output_stream_structure: Parser.stream_structure(),
+            stable_reprefixing?: boolean()
           }
-
-  defstruct scheme_parser_state: SchemeParser.new()
+  @enforce_keys [:input_stream_structure, :output_stream_structure]
+  defstruct @enforce_keys ++
+              [scheme_parser_state: SchemeParser.new(), stable_reprefixing?: true]
 
   @doc """
   Returns a structure holding a clear NALu parser state.
```

Comment:
we can add some description of that `stable_reprefixing` option

Reply:
is that `stable_(re)prefixing` mode used somewhere else than in the tests?

Reply:
no